### ISSUE TITLE
youtube-dl doesn't see the video file if page contains YouTube trailer

### ISSUE THREAD

**Issue by rhazor** [rg3/youtube-dl#15401]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.01.21*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.01.21**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
youtube-dl -v https://www.kinofondas.lt/filmas/laikinai-/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.kinofondas.lt/filmas/laikinai-/']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2018.01.18
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.16299
[debug] exe versions: ffmpeg N-86950-g1bef008, ffprobe N-86950-g1bef008
[debug] Proxy map: {}
[generic] laikinai-: Requesting header
WARNING: Falling back on generic information extractor.
[generic] laikinai-: Downloading webpage
[generic] laikinai-: Extracting information
[download] Downloading playlist: Laikinai
[generic] playlist Laikinai: Collected 1 video ids (downloading 1 of them)
[download] Downloading video 1 of 1
[youtube] XCm5U-fneMo: Downloading webpage
[youtube] XCm5U-fneMo: Downloading video info webpage
[youtube] XCm5U-fneMo: Extracting video information
[debug] Default format spec: bestvideo+bestaudio/best
WARNING: Requested formats are incompatible for merge and will be merged into mkv.
[debug] Invoking downloader on 'https://r8---sn-uxv-8ovl.googlevideo.com/videoplayback?itag=136&pl=20&mime=video%2Fmp4&keepalive=yes&mm=31&mn=sn-uxv-8ovl&aitags=133%2C134%2C135%2C136%2C160%2C242%2C243%2C244%2C247%2C278&gir=yes&requiressl=yes&mt=1516740784&mv=m&ei=EaFnWreeMtHb7ATevpzABw&ms=au&ip=<MY_IP_EDITED>&key=yt6&lmt=1425717181198456&dur=64.833&expire=1516762481&clen=12092425&beids=%5B9466592%5D&id=o-AIrFBjioNpOUWgiP4XgLbtEzJEdpb9JfQvGv-S0as8k7&source=youtube&initcwndbps=873750&sparams=aitags%2Cclen%2Cdur%2Cei%2Cgir%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Ckeepalive%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Crequiressl%2Csource%2Cexpire&ipbits=0&signature=53AC0D6CE479425418ACC6DFD053D239913A655D.89313904B6B52353C80B0C3CD6A7A986A002FF85&ratebypass=yes'
[download] Destination: LAIKINAI trailer-XCm5U-fneMo.f136.mp4
[download] 100% of 11.53MiB in 00:18
[debug] Invoking downloader on 'https://r8---sn-uxv-8ovl.googlevideo.com/videoplayback?itag=251&pl=20&mime=audio%2Fwebm&keepalive=yes&mm=31&mn=sn-uxv-8ovl&gir=yes&requiressl=yes&mt=1516740784&mv=m&ei=EaFnWreeMtHb7ATevpzABw&ms=au&ip=<MY_IP_EDITED>&key=yt6&lmt=1432463668288276&dur=64.881&expire=1516762481&clen=963948&beids=%5B9466592%5D&id=o-AIrFBjioNpOUWgiP4XgLbtEzJEdpb9JfQvGv-S0as8k7&source=youtube&initcwndbps=873750&sparams=clen%2Cdur%2Cei%2Cgir%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Ckeepalive%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Crequiressl%2Csource%2Cexpire&ipbits=0&signature=A86794240E36291A9F1566749AB552479FBF50F5.82B89C05D2151EDC8E8B82121D6ECA8E7EAC8D37&ratebypass=yes'
[download] Destination: LAIKINAI trailer-XCm5U-fneMo.f251.webm
[download] 100% of 941.36KiB in 00:04
[ffmpeg] Merging formats into "LAIKINAI trailer-XCm5U-fneMo.mkv"
[debug] ffmpeg command line: ffmpeg -y -i "file:LAIKINAI trailer-XCm5U-fneMo.f136.mp4" -i "file:LAIKINAI trailer-XCm5U-fneMo.f251.webm" -c copy -map "0:v:0" -map "1:a:0" "file:LAIKINAI trailer-XCm5U-fneMo.temp.mkv"
Deleting original file LAIKINAI trailer-XCm5U-fneMo.f136.mp4 (pass -k to keep)
Deleting original file LAIKINAI trailer-XCm5U-fneMo.f251.webm (pass -k to keep)
[download] Finished downloading playlist: Laikinai
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://www.kinofondas.lt/filmas/laikinai-/ (click yellow play icon in the middle of the page)

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

e.g. this page has no YouTube trailer https://www.kinofondas.lt/filmas/filmas-8-minutes/ so youtube-dl downloads the movie the fine. Under --list-formats it lists one file with format mp4 and resolution unknown.

Same scenario, but a page with YouTube trailer included e.g. at https://www.kinofondas.lt/filmas/laikinai-/ and it downloads only the trailer from YouTube. It's not a coincidence, I tried multiple other pages that has no trailer and they all work, but as soon as there a page with YouTube trailer, the movie file gets ignored and only the trailer is downloaded, it also sees only YouTube formats under --list-formats.

tldr: somehow youtube trailer is blocking the video.

**Comment by rhazor** [comment]
--json-dump gives youtube only URLs to googlevideo.com and etc if the page contains YouTube or Vimeo trailers included. But if the page does not have any of those, then it gives right data. E.g.

`{"playlist_title": "8 minut\u0117s", "playlist": "8 minut\u0117s", "webpage_url": "https://www.kinofondas.lt/filmas/filmas-8-minutes/", "n_entries": 1, "requested_subtitles": null, "webpage_url_basename": "filmas-8-minutes", "extractor": "generic", "playlist_uploader_id": null, "playlist_uploader": null, "url": "https://www.kinofondas.lt/filmas.php?filmas=0d1258ef5d7c30c5b06ae71631b2ecb4", "subtitles": {}, "id": "filmas-8-minutes", "ext": "mp4", "format": "0 - unknown", "format_id": "0", "fulltitle": "8 minut\u0117s", "formats": [{"format_id": "0", "http_headers": {"Accept-Language": "en-us,en;q=0.5", "Cookie": "PHPSESSID=26gd8gi5fujmboiov17c9olhj5", "Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/47.0 (Chrome)"}, "url": "https://www.kinofondas.lt/filmas.php?filmas=0d1258ef5d7c30c5b06ae71631b2ecb4", "format": "0 - unknown", "ext": "mp4", "vcodec": null, "protocol": "https", "height": null}], "title": "8 minut\u0117s", "display_id": "filmas-8-minutes", "height": null, "playlist_id": "filmas-8-minutes", "http_headers": {"Accept-Language": "en-us,en;q=0.5", "Cookie": "PHPSESSID=26gd8gi5fujmboiov17c9olhj5", "Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7", "Accept-Encoding": "gzip, deflate", "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/47.0 (Chrome)"}, "extractor_key": "Generic", "playlist_index": 1, "_filename": "8 minut\u0117s-filmas-8-minutes.mp4", "thumbnails": [{"url": "img/videofonas.jpg", "id": "0"}], "protocol": "https", "vcodec": null, "thumbnail": "img/videofonas.jpg"}
`

Why none of this is fetched and ignored in pages that contain YouTube or Vimeo trailers? How do I fetch this part on those pages, how do I ignore YouTube or Vimeo so it would fetch the right thing? Because now somehow it's ignored by YouTube/Vimeo.

**Comment by dstftw** [comment]
This site is not supported.
